torch/csrc/jit/passes/python_print.cpp

```diff
@@ -402,26 +469,33 @@ struct PythonPrintPass {
   }
 
   // map from Value to how it should be printed at each use
-  std::unordered_map<Value*, std::string> value_names_;
+  std::unordered_map<Value*, std::shared_ptr<TaggedStringStream>> value_names_;
 
-  std::string useOf(Value* v) const {
+  // NB: we MUST pass around the shared pointers to these streams by value.
+  // There is an interaction in splitLongInlines where the string value for
+  // both the RHS and the LHS of an expression are live at the same time,
+  // however the value for the RHS is overwritten in the table.
+  std::shared_ptr<TaggedStringStream> useOf(Value* v) const {
     return value_names_.at(v);
   }
-  void assignValue(Value* v, const std::string& s) {
-    value_names_[v] = s;
+  void assignValue(Value* v, std::shared_ptr<TaggedStringStream> s) {
```

Comment:
This is a complicated case. There are two possible meanings for useOf -- it is a variable name, or it is a inlined definition. When it is a variable name, then typically we _do not_ want to change the source range. For instance on the line `a + b`, we want to report where the add occurred, _not_ where `a` is defined. But if `a` has the source range for where it is defined, then the entire expression will look like it occurs there.

If the value is an inlined definition, then we _do_ want to change the source range. We should probably handle these cases separately (with variable mappings taking a raw string, and inlined definitions taking a TaggedStringStream) in order to get this right.

Reply:
Does the python printer emit infix operators anywhere? I just tried your example, and since we emit a prefix expression for add, it works properly:

```
        class FooTest(torch.jit.ScriptModule):
            @torch.jit.script_method
            def forward(self, x, y):
                return x + y

        ft = FooTest()
        loaded = self.getExportImportCopy(ft)
        _, lineno = inspect.getsourcelines(FooTest)

        loaded(torch.rand(3, 4), torch.rand(30, 40))
```

```
The above operation failed in interpreter, with the following stack trace:
at code/tmpcoo1em5o.py:5:10
op_version_set = 1
def forward(self,
    x: Tensor,
    y: Tensor) -> Tensor:
  return torch.add(x, y, alpha=1)
         ~~~~~~~~~ <--- HERE
Compiled from code at test/test_jit.py:3377:24
            @torch.jit.script_method
            def forward(self, x, y):
                return x + y
                       ~~~~~ <--- HERE
"
```

Reply:
I fixed this anyway since it would help reduce the amount of data we have to serialize